app/models/custom_filter.rb

```diff
@@ -24,6 +24,29 @@ class CustomFilter < ApplicationRecord
   enum filter_type: { conversation: 0, contact: 1, report: 2 }
   validate :validate_number_of_filters
 
+  def records_count
+    fetch_record_count_from_redis
+  end
+
+  def filter_records
+    Conversations::FilterService.new(query.with_indifferent_access, user, account).perform
+  end
+
+  def set_record_count_in_redis
+    records = filter_records
+    Redis::Alfred.set(filter_count_key, records[:count][:all_count])
+  end
+
+  def fetch_record_count_from_redis
+    number_of_records = Redis::Alfred.get(filter_count_key)
+    SyncCustomFilterCountJob.perform_later(self) if number_of_records.nil?
```

Comment:
Should we keep this job synchronous? If the records are nil or outdated, we calculate on the fly and return. This ensures we always share the correct values

Reply:
It would significantly overload the filter endpoint, as the calculation would be pretty time-consuming. + we would be iterating over each filter and will be calculating for each right?

Reply:
> We would be iterating over each filter and will be calculating for each, right?

Yes, that's why we didn't keep it in the synchronous job, it's getting updated on every new conversation update.

Reply:
Noted, in that case this looks good